board.js

```diff
@@ -224,9 +233,11 @@ export default class Board {
     let hasLineOfSight = false;
     let sourceCorner = null;
     let targetCorners = null;
+    let minDistSquaredSum = null;
```

Comment:
you could set this to Number.MAX_VALUE so then it is only a number instead of a ?number, simplifying your if check below